ocaml/libs/clock/timer.ml

```diff
@@ -0,0 +1,53 @@
+type t = {start: Ptime.t; elapsed: Mtime_clock.counter; timeout: Mtime.Span.t}
+
+type remaining = Spare of Mtime.Span.t | Excess of Mtime.Span.t
+
+let is_shorter a ~than:b = Mtime.Span.compare a b < 0
+
+let is_longer a ~than:b = Mtime.Span.compare a b > 0
+
+let start ~timeout =
+  {start= Ptime_clock.now (); elapsed= Mtime_clock.counter (); timeout}
+
+let timeout {timeout; _} = timeout
+
+let elapsed t = Mtime_clock.count t.elapsed
+
+let remaining t =
+  let elapsed = Mtime_clock.count t.elapsed in
+  let difference = Mtime.Span.abs_diff t.timeout elapsed in
+  if Mtime.Span.compare t.timeout elapsed > 0 then
+    Spare difference
+  else
+    Excess difference
+
+let expired t = match remaining t with Spare _ -> false | Excess _ -> true
+
+let deadline_of t =
```

Comment:
I think XAPI uses deadlines the other way around, i.e. they are relative deadlines from when an operation started. (well currently they are based on gettimeofday, but I'm fixing some of those in my epoll PR, they appear to be absolute deadlines, but really are relative, because they are gettimeofday + delta).
So we don't need to use or convert to absolute timestamps here, and thus we can drop the Ptime.t from the type.

Reply:
I've removed the deadlines